### ISSUE TITLE
Configuring pre-processors

### ISSUE THREAD

**Issue by dvberkel** [rust-lang-nursery/mdBook#622]
The documentation on [pre-processors](https://rust-lang-nursery.github.io/mdBook/for_developers/preprocessors.html) has left me with some questions. Specifically how to add/configure them. I only see one way to do that, but I could well be mistaken.

It is clear to me that you could create a custom build script with the mdBook library, adding your custom pre-processors with the existing pre-processors in the order that you want them in. It this what I should expect? Or am I missing something?

**Comment by Michael-F-Bryan** [comment]
As the preprocessors feature is quite new we thought we'd initially start off small and then extend things depending on people's needs and requests. 

At the moment you can specify a list of preprocessor names in your `book.toml` and then `mdbook` will go through that list looking for pre-defined preprocessors. In the long term I'd like to extend it to be similar to how configuration works for backends, so you could create a `build.preprocessor.foo` table and pass in extra options that way. I'd also like to allow people to create their own preprocessors, which we could probably implement by shelling out to the specified program on their computer.

TL;DR: Configuring preprocessors isn't possible *yet*, but if you want to you can pull in the `mdbook` crate and create your own `mdbook` program.

**Comment by dvberkel** [comment]
@Michael-F-Bryan Thanks for clearing that up. That answers my question.
